lib/tasks/ldap.rake

```
@@ -1,107 +1,161 @@
-       end
-
-      raise "Missing HOST" if ENV['HOST'].blank?
-      host = ENV['HOST']
-
-      raise "Missing PORT" if ENV['PORT'].blank?
-      port = ENV['PORT']
-
-      encryption = ENV['ENCRYPTION'].blank? ? nil : ENV['ENCRYPTION']
-
-      ssl_version = ENV['SSL_VERSION'].blank? ? nil : ENV['SSL_VERSION']
-
-      raise "Missing CONNECTION_USER" if ENV['CONNECTION_USER'].blank?
-      connection_user = ENV['CONNECTION_USER']
-
-      raise "Missing CONNECTION_PASSWORD" if ENV['CONNECTION_PASSWORD'].blank?
-      connection_password = ENV['CONNECTION_PASSWORD']
-
-      raise "Missing USER_ID_FIELD" if ENV['USER_ID_FIELD'].blank?
-      user_id_field = ENV['USER_ID_FIELD']
+      else
+        organization = Carto::Organization.find(ENV['ORGANIZATION_ID'])
+      end
+      ldap = ldap_configuration_from_environment(organization)
 
-      raise "Missing USERNAME_FIELD" if ENV['USERNAME_FIELD'].blank?
-      username_field = ENV['USERNAME_FIELD']
+      unless ldap.valid?
+        missing = ldap.errors.keys.reject { |k| k == :domain_bases_list }
+        raise "Missing: " + missing.join(', ').upcase
+      end
 
-      email_field = ENV['EMAIL_FIELD'].blank? ? nil : ENV['EMAIL_FIELD']
+      if ldap.save
+        puts "LDAP configuration created with id: #{ldap.id}"
+      else
+        puts "Error saving LDAP configuration"
+      end
+    end
 
-      raise "Missing DOMAIN_BASES" if ENV['DOMAIN_BASES'].blank?
-      domain_bases = ENV['DOMAIN_BASES'].split(Carto::Ldap::Configuration::DOMAIN_BASES_SEPARATOR)
+    desc "Deletes existing LDAP Configuration entries"
+    task :reset_ldap_configuration, [] => :environment do |_t, _args|
+      Carto::Ldap::Configuration.delete_all
+    end
+  end
 
-      raise "Missing USER_OBJECT_CLASS" if ENV['USER_OBJECT_CLASS'].blank?
-      user_object_class = ENV['USER_OBJECT_CLASS']
+  private
+
+  def ldap_configuration_from_environment(organization)
+    # Mandatory: connection parameters
```

Maybe we should raise if environment variables are not present... I guess it will fail further down the stack anyway